src/tools/debugger/nodes/BreakpointNode.java

```diff
@@ -0,0 +1,57 @@
+package tools.debugger.nodes;
+
+import com.oracle.truffle.api.Assumption;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.VM;
+import tools.SourceCoordinate;
+import tools.debugger.session.BreakpointEnabling;
+import tools.debugger.session.Breakpoints.MessageBreakpointType;
+
+/**
+ * Node to represent a breakpoint at the AST level.
+ * It has two possible states, enable or disable.
+ */
+public abstract class BreakpointNode extends AbstractBreakpointNode {
+    protected final BreakpointEnabling<?> breakpoint;
```

Comment:
Please make this a proper generic type. Replace `?` with `T`. And add `T` as a generic argument to `BreakpointNode`